rs/test_utilities/consensus/src/idkg.rs

```diff
@@ -78,18 +84,128 @@ pub fn empty_response() -> ConsensusResponse {
     ConsensusResponse::new(CallbackId::from(0), Payload::Data(vec![]))
 }
 
-fn fake_signature_request_args(key_id: MasterPublicKeyId, height: Height) -> ThresholdArguments {
+pub fn fake_pre_signature_stash(key_id: &IDkgMasterPublicKeyId, size: u64) -> PreSignatureStash {
+    let rng = &mut reproducible_rng();
+    let env = CanisterThresholdSigTestEnvironment::new(4, rng);
+    let (dealers, receivers) =
+        env.choose_dealers_and_receivers(&IDkgParticipants::AllNodesAsDealersAndReceivers, rng);
+    let alg = key_id_to_algorithm_id(key_id);
+    let key = generate_key_transcript(&env, &dealers, &receivers, alg, rng);
+    let mut pre_signatures = BTreeMap::new();
+    for i in 0..size {
+        let pre_signature = match key_id.inner() {
+            MasterPublicKeyId::Ecdsa(_) => {
+                let pre_sig =
+                    generate_ecdsa_presig_quadruple(&env, &dealers, &receivers, alg, &key, rng);
+                PreSignature::Ecdsa(Arc::new(pre_sig))
+            }
+            MasterPublicKeyId::Schnorr(_) => {
+                let blinder_params =
+                    setup_unmasked_random_params(&env, alg, &dealers, &receivers, rng);
+                let blinder_transcript = env
+                    .nodes
+                    .run_idkg_and_create_and_verify_transcript(&blinder_params, rng);
+                PreSignature::Schnorr(Arc::new(
+                    SchnorrPreSignatureTranscript::new(blinder_transcript).unwrap(),
+                ))
+            }
+            MasterPublicKeyId::VetKd(_) => panic!("Not an IDkgMasterPublicKeyId"),
+        };
+        pre_signatures.insert(PreSigId(i), pre_signature);
+    }
+    PreSignatureStash {
+        key_transcript: Arc::new(key),
+        pre_signatures,
+    }
+}
+
+fn ecdsa_curve_to_algorithm_id(curve: EcdsaCurve) -> AlgorithmId {
+    match curve {
+        EcdsaCurve::Secp256k1 => AlgorithmId::ThresholdEcdsaSecp256k1,
+    }
+}
+
+fn schnorr_algorithm_to_algorithm_id(alg: SchnorrAlgorithm) -> AlgorithmId {
+    match alg {
+        SchnorrAlgorithm::Bip340Secp256k1 => AlgorithmId::ThresholdSchnorrBip340,
+        SchnorrAlgorithm::Ed25519 => AlgorithmId::ThresholdEd25519,
+    }
+}
+
+fn key_id_to_algorithm_id(key_id: &IDkgMasterPublicKeyId) -> AlgorithmId {
```

Comment:
There is already a `key_id_to_algorithm_id` function in `consensus/idkg`.
I know that we can't use that one here, since it is in an upstream crate, and we can't use this one there, since it is only a `dev` dependency.

I think this means, that we should implement (in a separate PR) this function in the `types` crate, since both `IDkgMasterPublicKeyId` and `AlgorithmId` live there.

What do you think?